src/FluentAssertions.Analyzers/Utilities/FluentAssertionsAnalyzer.cs

```diff
@@ -23,35 +23,26 @@ public sealed override void Initialize(AnalysisContext context)
         {
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
-            context.RegisterCodeBlockAction(AnalyzeCodeBlock);
+            context.RegisterSyntaxNodeAction(AnalyzeExpressionStatementSyntax, SyntaxKind.ExpressionStatement, SyntaxKind.ArrowExpressionClause);
         }
 
-        private void AnalyzeCodeBlock(CodeBlockAnalysisContext context)
+        private void AnalyzeExpressionStatementSyntax(SyntaxNodeAnalysisContext context)
         {
-            var method = context.CodeBlock as MethodDeclarationSyntax;
-            if (method == null) return;
-
-            if (!ShouldAnalyzeMethod(method)) return;
-
-            if (method.Body != null)
+            ExpressionSyntax expression = context.Node.Kind() switch
+            {
+                SyntaxKind.ExpressionStatement => ((ExpressionStatementSyntax)context.Node).Expression,
+                SyntaxKind.ArrowExpressionClause => ((ArrowExpressionClauseSyntax)context.Node).Expression,
+                _ => null
```

Comment:
As this can not occur, why not throw?

Reply:
oh, because of the way I'm calling the `RegisterSyntaxNodeAction` with the `SyntaxKind` arguments?

Reply:
Indeed. So Your code is only hit for those two syntax kinds.